base/applications/osk/main.c

```diff
@@ -57,6 +58,47 @@ int OSK_SetImage(int IdDlgItem, int IdResource)
     return TRUE;
 }
 
+/***********************************************************************
+ *
+ *          OSK_WarningProc
+ *
+ *  Function handler for the warning dialog box on startup
+ */
+INT_PTR CALLBACK OSK_WarningProc(HWND hDlg, UINT Msg, WPARAM wParam, LPARAM lParam)
+{
+    UNREFERENCED_PARAMETER(lParam);
+
+    switch (Msg)
+    {
+        case WM_INITDIALOG:
+        {
+            return TRUE;
+        }
+
+        case WM_COMMAND:
+        {
+            switch (LOWORD(wParam))
+            {
+                case IDC_SHOWWARNINGCHECK:
+                {
+                    Globals.bShowWarning = !CheckDlgButton(hDlg, IDC_SHOWWARNINGCHECK, BST_CHECKED);
+                    break;
```

Comment:
Is returning `FALSE` correct here?

Reply:
You must be speaking for `!CheckDlgButton`, right? If so yes it's correct. But it looks like it is impossible to un-check the button afterwards due to `BST_CHECKED`. I've to fix that in a future commit.